stellar-dotnet-sdk-test/RevokeSponsorshipTest.cs

```diff
@@ -0,0 +1,441 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NUnit.Framework;
+using stellar_dotnet_sdk;
+using stellar_dotnet_sdk.responses;
+using stellar_dotnet_sdk.responses.results;
+using Assert = Microsoft.VisualStudio.TestTools.UnitTesting.Assert;
+using Claimant = stellar_dotnet_sdk.Claimant;
+
+namespace stellar_dotnet_sdk_test;
+
+[TestClass]
+public class RevokeSponsorshipTest
+{
+    private readonly Server _server = new("https://horizon-testnet.stellar.org");
+    private string _sponsoredId = "GC3TDMFTMYZY2G4C77AKAVC3BR4KL6WMQ6K2MHISKDH2OHRFS7CVVEAF";
+    private string _sponsoringId = "GARRDNS77ZSI6PPXRBWTHIVX4RS2ULVBKNJXFRV77AZUNLDUNV2NAHJA";
+
+    private Asset _assetA =
+        new AssetTypeCreditAlphaNum4("AAA", "GARRDNS77ZSI6PPXRBWTHIVX4RS2ULVBKNJXFRV77AZUNLDUNV2NAHJA");
+
+    private KeyPair _sponsoringAccount =
+        KeyPair.FromSecretSeed("SDR4PTKMR5TAQQCL3RI2MLXXSXQDIR7DCAONQNQP6UCDZCD4OVRWXUHI");
+
+    private KeyPair _sponsoredAccount =
+        KeyPair.FromSecretSeed("SDBNUIC2JMIYKGLJUFI743AQDWPBOWKG42GADHEY3FQDTQLJADYPQZTP");
+
+    private const string DataName = "my secret";
+
+    [TestInitialize]
+    public async Task Setup()
+    {
+        Network.UseTestNetwork();
+        // Generates a new _accountId in case the testnet has been reset
+        try
+        {
+            await _server.Accounts.Account(_sponsoringId);
+        }
+        catch (AccountNotFoundException)
+        {
+            _sponsoringAccount = await CreateNewRandomAccountOnTestnet();
```

Comment:
Why create a random account? This will result in creating a new account every time right? Check the `_sponsoringKeyPair` and if that is not found just create a account for that specific key pair.